inttest/toolsuite/operations/planapply.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 k0s authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+
+	ts "github.com/k0sproject/k0s/inttest/toolsuite"
+	tsutil "github.com/k0sproject/k0s/inttest/toolsuite/util"
+	apv1beta2 "github.com/k0sproject/k0s/pkg/apis/autopilot.k0sproject.io/v1beta2"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type PlanBuilder func(output tsutil.TerraformOutputMap) (*apv1beta2.Plan, error)
+
+// PlanApplyOperation is an Operation that will read output values from terraform,
+// build a user-provided Plan, and apply it via kubectl.
+func PlanApplyOperation(ctx context.Context, builder PlanBuilder) ts.ClusterOperation {
+	return func(data ts.ClusterData) error {
+		// The the terraform binary, error if missing
+		tfpath, err := exec.LookPath("terraform")
+		if err != nil {
+			return fmt.Errorf("unable to find terraform binary: %w", err)
+		}
+
+		// Initialize terraform to read the state output values.
+		tf, err := tfexec.NewTerraform(data.DataDir, tfpath)
+		if err != nil {
+			return fmt.Errorf("failed to initialize terraform: %w", err)
+		}
+
+		tf.SetStdout(os.Stdout)
```

Comment:
Is that teeing to stdout? The `tf.Output` command below looks like it would also return the output as a string.

Reply:
Yes, teeing to stdout. The `tf.Output()` below returns a terraform-specific map of json encoded values that operators will pick and choose from.